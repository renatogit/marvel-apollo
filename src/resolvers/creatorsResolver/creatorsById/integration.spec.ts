import DataSourcesCreatorByIdAPI from '@/dataSources/creators';
import CreatorByIdResolver from '.';
import ServerGetCreatorById from '@/configs/test';
import QUERY_GET_CREATOR_BY_ID from '@query/creators/queryCreatorById';
import MOCK_GET_CREATOR_BY_ID from '@mock/creators/mockCreatorById';

describe('resolvers/creatorByIdResolver/integration', () => {
	const variables = {creatorsId: 'fake-code'};

	it('fetches creatorById and generate snapshots', async () => {
		const args = {
			dataSources: DataSourcesCreatorByIdAPI,
			resolver: CreatorByIdResolver.Query.creatorById,
			mock: MOCK_GET_CREATOR_BY_ID,
			variables,
			entity: 'creators',
			typeQuery: 'creatorById',
		};

		const {server} = await ServerGetCreatorById(args);

		const {body} = await server.executeOperation({
			query: QUERY_GET_CREATOR_BY_ID,
			variables,
		});

		expect(body).toMatchSnapshot();
	});
});
