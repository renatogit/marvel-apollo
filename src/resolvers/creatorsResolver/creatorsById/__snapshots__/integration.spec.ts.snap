// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolvers/creatorByIdResolver/integration fetches creatorById and generate snapshots 1`] = `
{
  "kind": "single",
  "singleResult": {
    "data": {
      "creatorById": {
        "attributionHTML": "mock data",
        "attributionText": "mock data",
        "code": 200,
        "copyright": "mock data",
        "data": {
          "count": 1,
          "limit": 10,
          "offset": 0,
          "results": [
            {
              "comics": [
                {
                  "collections": [
                    {
                      "name": "mock data",
                      "resourceURI": "mock data",
                    },
                  ],
                  "dates": [
                    {
                      "date": "mock data",
                      "type": "mock data",
                    },
                  ],
                  "description": "mock data",
                  "id": 15902,
                  "images": [
                    {
                      "extension": "mock data",
                      "path": "mock data",
                    },
                  ],
                  "modified": "mock data",
                  "pageCount": 0,
                  "prices": [
                    {
                      "price": 9.99,
                      "type": "mock data",
                    },
                  ],
                  "thumbnail": {
                    "extension": "mock data",
                    "path": "mock data",
                  },
                  "title": "mock data",
                  "urls": [
                    {
                      "type": "mock data",
                      "url": "mock data",
                    },
                  ],
                },
              ],
              "events": [
                {
                  "description": "mock data",
                  "end": "mock data",
                  "id": 1,
                  "modified": "mock data",
                  "start": "mock data",
                  "thumbnail": {
                    "extension": "mock data",
                    "path": "mock data",
                  },
                  "title": "mock data",
                  "urls": [
                    {
                      "type": "mock data",
                      "url": "mock data",
                    },
                  ],
                },
              ],
              "fullName": "mock data",
              "id": 13970,
              "modified": "mock data",
              "series": [
                {
                  "description": "mock data",
                  "endYear": 2000,
                  "id": 378,
                  "modified": "mock data",
                  "rating": "mock data",
                  "startYear": 1999,
                  "thumbnail": {
                    "extension": "mock data",
                    "path": "mock data",
                  },
                  "title": "mock data",
                  "urls": [
                    {
                      "type": "mock data",
                      "url": "mock data",
                    },
                  ],
                },
              ],
              "stories": [
                {
                  "description": "mock data",
                  "id": 25655,
                  "modified": "mock data",
                  "thumbnail": {
                    "extension": "mock data",
                    "path": "mock data",
                  },
                  "title": "mock data",
                  "type": "mock data",
                },
              ],
              "suffix": "mock data",
              "thumbnail": {
                "extension": "mock data",
                "path": "mock data",
              },
              "urls": [
                {
                  "type": "mock data",
                  "url": "mock data",
                },
              ],
            },
          ],
          "total": 1,
        },
        "etag": "mock data",
        "status": "mock data",
      },
    },
    "errors": undefined,
  },
}
`;
